### PR TITLE
Add pushing images to ghcr.io for release builds

### PR DESCRIPTION
**Pull request by pmalek-sumo**
###### Description

Add pushing images to ghcr.io

###### Testing performed

- [ ] ci/build.sh
- [ ] Redeploy fluentd and fluentd-events pods
- [ ] Confirm events, logs, and metrics are coming in

### CODE REVIEW

**Review by perk-sumo — APPROVED** [review]
👍

File: ci/_build_functions.sh [hunk 1]
```diff
@@ -69,6 +69,15 @@ function push_docker_image() {
...
   docker push "${docker_tag}:${version}"
+
+  if [[ -n "${CR_PAT}" && -n "${CR_OWNER}" ]]; then
+    echo "${CR_PAT}" | docker login ghcr.io -u "${CR_OWNER}" --password-stdin
+    readonly GHCR_IO_DOCKER_TAG="ghcr.io/${docker_tag}:${version}"
+    echo "Tagging docker image ${docker_tag}:local with ${GHCR_IO_DOCKER_TAG}..."
+    docker tag "${docker_tag}:local" "${GHCR_IO_DOCKER_TAG}"
+    echo "Pushing docker image ${GHCR_IO_DOCKER_TAG}..."
+    docker push "${GHCR_IO_DOCKER_TAG}"
+  fi
```
Comment (sumo-drosiek) [inline ci/_build_functions.sh:80]: Can we move it to the function and make it general by passing login, password, registry?

Comment (sumo-drosiek) [reply ci/_build_functions.sh:80]: and `${docker_tag}`

Comment (sumo-drosiek) [reply ci/_build_functions.sh:80]: I mean, push_docker_image should push image to a registry and you can eventually create `push_docker_images` which will check for gcrio, ecr, dockerhub, quay.io, etc. and will run `push_docker_image` with arguments for every of the registry

**Review by sumo-drosiek — APPROVED** [review]
We can postpone my request. It should be done in #1220

**Pull request merged**